.github/scripts/publish-rhel.sh

```diff
@@ -1,6 +1,9 @@
 #!/bin/bash
 
-set -euo pipefail ${RUNNER_DEBUG:+-x}
+set -o errexit -o nounset -o pipefail ${RUNNER_DEBUG:+-x}
+
+# shellcheck source=../.github/scripts/abstract-simple-smoke-test.sh
+. .github/scripts/abstract-simple-smoke-test.sh
```

Comment:
Why do we need it?

Reply:
For the `echoerr` function.

Reply:
It's confusing that we have a test code in the production script. I would suggest either inline this function as it's very small or move it to another file `echo.functions.sh`/`log.functions.sh`?

Reply:
> It's confusing that we have a test code in the production script.

Good point.

> I would suggest either inline this function as it's very small or move it to another file `echo.functions.sh`/`log.functions.sh`?

For now I've duplicated it.
_Ideally_ it should be externalised - but [as `hazelcast-packaging` directly references the script without any other context](https://github.com/hazelcast/hazelcast-packaging/blob/9b5cd38369c7fdf883f6e9a58bb29c7e1a46b615/check-hazelcast-health.sh#L8), a larger change is required to support that.